#VRML_SIM R2019b utf8
WorldInfo {
}
Viewpoint {
  orientation 0.9961666532542606 0.06619075233216687 -0.05719076192802217 5.635623824554719
  position -0.06760540693980638 2.9124991491717642 3.835345250019569
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
DEF OMNI_WHEELS Robot {
  translation 0 0.06 0
  children [
    DEF SOLID1 Solid {
      children [
        DEF WHEEL1 HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            PositionSensor {
              name "pos_sensor1"
              resolution 0.005235983
            }
            RotationalMotor {
              name "right_wheel"
              maxVelocity 52.35
              maxTorque 0.59
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 1 0
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r1 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        DEF SMALL_WHEEL_SHAPE Shape {
                          appearance PBRAppearance {
                            baseColor 1 0.752941 0.796078
                            roughness 1
                            metalness 0
                          }
                          geometry Cylinder {
                            height 0.02
                            radius 0.01
                          }
                        }
                      ]
                    }
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r2 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r3 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r4 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  children [
                    DEF r5 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  children [
                    DEF r6 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r7 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r8 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF WHEEL_TRANS Transform {
                translation 0 0 0.02
                rotation 1 0 0 1.5708
                children [
                  DEF WHEEL_SHAPE Shape {
                    appearance PBRAppearance {
                      baseColor 0.678431 0.847059 0.901961
                      roughness 1
                      metalness 0
                    }
                    geometry Cylinder {
                      height 0.02
                      radius 0.05
                      subdivision 20
                    }
                  }
                ]
              }
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "right_wheel"
      boundingObject DEF DUMMY Sphere {
        radius 0.01
      }
      physics Physics {
      }
    }
    DEF SOLID2 Solid {
      rotation 0 1 0 2.09
      children [
        DEF WHEEL2 HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            PositionSensor {
              name "pos_sensor2"
              resolution 0.005235983
            }
            RotationalMotor {
              name "left_wheel"
              maxVelocity 52.35
              maxTorque 0.59
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 1 0.00240580999995935
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 0
                  children [
                    USE r1
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r2
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation 1 0 0 0
                  children [
                    USE r3
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r4
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  children [
                    USE r5
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  children [
                    USE r6
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r7
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r8
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              USE WHEEL_TRANS
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "left_wheel"
      boundingObject USE DUMMY
      physics Physics {
      }
    }
    DEF SOLID3 Solid {
      rotation 0 1 0 4.18
      children [
        DEF WHEEL3 HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            PositionSensor {
              name "pos_sensor3"
              resolution 0.005235983
            }
            RotationalMotor {
              name "back_wheel"
              maxVelocity 52.35
              maxTorque 0.59
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 1 0.0039094400000208316
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 0
                  children [
                    USE r1
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r2
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation 1 0 0 0
                  children [
                    USE r3
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r4
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  children [
                    USE r5
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  children [
                    USE r6
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r7
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    USE r8
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              USE WHEEL_TRANS
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "back_wheel"
      boundingObject USE DUMMY
      physics Physics {
      }
    }
    DEF BODY_TRANS Transform {
      translation 0 0.03 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.564706 0.933333 0.564706
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.15
            radius 0.1
            subdivision 18
          }
        }
      ]
    }
    DEF RED_SENSOR DistanceSensor {
      translation 0.0941644 0.08 0.0412099
      rotation 0 1 0 0.08720030610042528
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.986603 0.12834399999999999 0.047744999999999996
          }
          geometry Box {
            size 0.02 0.01 0.01
          }
        }
      ]
      name "dist_left_sensor1"
      lookupTable [
        0 0 0
        0.02 0 0
        5 1659 0
      ]
    }
    DEF BLUE_SENSOR DistanceSensor {
      translation 0.070247354014814 0.08 0.07480324640448982
      rotation 0 -1 0 0.08726030610042529
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0.998199
          }
          geometry Box {
            size 0.02 0.01 0.01
          }
        }
      ]
      name "dist_right_sensor2"
      lookupTable [
        0 0 0
        0.02 0 0
        5 1659 0
      ]
    }
  ]
  boundingObject USE BODY_TRANS
  physics Physics {
  }
  controller "controller1"
}
Solid {
  translation 1.46 0.15 -0.74
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.242023 1 0.9991
      }
      geometry DEF G_BOX1 Box {
        size 0.3 0.3 0.3
      }
    }
  ]
  boundingObject USE G_BOX1
  physics Physics {
  }
}
Solid {
  translation 0.72 0.15 1.3399999999999999
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.242023 1 0.9991
      }
      geometry DEF G_BOX2 Box {
        size 0.3 0.3 0.3
      }
    }
  ]
  name "solid(1)"
  boundingObject USE G_BOX2
  physics Physics {
  }
}
Solid {
  translation 1.146232 0.15 0.524774
  rotation 0 1 0 1.5707959999999999
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.242023 1 0.9991
      }
      geometry DEF G_BOX3 Box {
        size 0.3 0.3 0.3
      }
    }
  ]
  name "solid(2)"
  boundingObject USE G_BOX3
  physics Physics {
  }
}
